app/src/main/java/com/sunasterisk/getitdone/data/source/local/dao/TaskDAOImpl.kt

```diff
@@ -0,0 +1,189 @@
+import com.sunasterisk.getitdone.utils.Constants.STATUS_NOT_COMPLETE
+import com.sunasterisk.getitdone.utils.Constants.TRUE
+
+class TaskDAOImpl private constructor(appDatabase: AppDatabase) : TaskDAO {
+
+    private val database = appDatabase.writableDatabase
+
+    override fun getAllTasksByListId(listId: Int): List<Task> {
+        return getTaskInSpecificCondition(Task.LIST_ID, listId.toString())
+    }
+
+    override fun getTaskById(id: Int): Task {
+        val cursor =
+            database.query(
+                Task.TABLE_NAME,
+                null,
+                Task.ID,
+                arrayOf(id.toString()),
+                null,
+                null,
+                null
+            )
+        cursor.close()
+        return getTaskFromCursor(cursor)
+    }
+
+    override fun getCompletedTasks(): List<Task> {
+        val cursor =
+            database.query(
+                Task.TABLE_NAME,
+                null,
+                Task.STATUS,
+                arrayOf(STATUS_COMPLETED),
+                null,
+                null,
+                null
+            )
+
+        val list = mutableListOf<Task>()
+        if (cursor.moveToFirst()) {
+            while (!cursor.isAfterLast) {
+                list.add(getTaskFromCursor(cursor))
+                cursor.moveToNext()
+            }
+        }
+        cursor.close()
+        return list
+    }
+
+    override fun getUnCompletedTasks(): List<Task> {
+        val cursor =
+            database.query(
+                Task.TABLE_NAME,
+                null,
+                Task.STATUS,
+                arrayOf(STATUS_NOT_COMPLETE),
+                null,
+                null,
+                null
+            )
+
+        val list = mutableListOf<Task>()
+        if (cursor.moveToFirst()) {
+            while (!cursor.isAfterLast) {
+                list.add(getTaskFromCursor(cursor))
+                cursor.moveToNext()
+            }
+        }
+        cursor.close()
+        return list
+    }
+
+    override fun searchTasksByTitle(title: String): List<Task> {
+        return getTaskInSpecificCondition(Task.TITLE, title)
+    }
+
+    override fun getTaskInMyDay(): List<Task> {
+        return getTaskInSpecificCondition(Task.IS_IN_MY_DAY, "$TRUE")
+    }
+
+    override fun getImportantTasks(): List<Task> {
+        return getTaskInSpecificCondition(Task.IS_IMPORTANT, "$TRUE")
+    }
+
+    override fun addNewTask(task: Task): Boolean {
+        val values = ContentValues().apply {
+            put("LIST_ID", task.listId)
+            put("TITLE", task.title)
+            put("DESCRIPTION", task.description)
+            put("TIME_REMINDER", task.timeReminder)
+            put("IS_IMPORTANT", task.isImportant)
+            put("IS_IN_MY_DAY", task.isInMyDay)
+            put("COLOR", task.color)
+            put("STATUS", task.status)
+            put("TIME_CREATED", task.timeCreated)
+        }
```

Comment:
Chuyển vào trong model luôn 
```suggestion
        val values = task.getContentValues()
```